### ISSUE TITLE
Not generate Path Params if parameters are defined in path dictionary

### ISSUE THREAD

**Issue by dangthaison91** [yonaskolb/SwagGen#25]
I noticed that you didn't parse Path parameters in class Endpoint. Below is Instagram API I got from Swagger Editor.

Solution: Should we replace `path: String` by `path: Path` where `Path` is a class/struct?
OTW, I was wonder why do you use `class` instead of `struct`. Maybe for creating other generators for other languages by inheritance?

YAML:
```yaml
paths:
  /users/{user-id}:
    parameters:
      - $ref: '#/parameters/user-id'
    get:
      security:
        - key: []
        - oauth:
          - basic
      tags:
        - Users
      description: Get basic information about a user.
     
  /media/{media-id}:
    parameters:
      - name: media-id
        in: path
        description: The media ID
        type: integer
        required: true
    get:
      tags:
        - Media
      description: |
        Get information about a media object.
        The returned type key will allow you to differentiate between `image`
        and `video` media.
```

**Comment by yonaskolb** [comment]
Hi @dangthaison91 
I'm not sure what you mean? Path parameters are replaced within the actual request when the url request is created https://github.com/yonaskolb/SwagGen/blob/master/Templates/Swift/request.swift#L73

In terms of the struct vs classes, which classes do you mean? If you're talking about the classes in the SwaggerSpec, it's because references, names and other properties are resolved in a spec in a second pass. This would be possible with structs, but it was easier to edit them with classes at the time without accidentally leaving a mutated struct behind.

**Comment by yonaskolb** [comment]
Here's an example of the path being replaced in an actual generated operation https://github.com/yonaskolb/SwagGen/blob/master/Specs/petstore/generated/Swift/Sources/API/Requests/Pets/ShowPetById.swift#L41

**Comment by dangthaison91** [comment]
Have you tried my YAML sample above @yonaskolb ? I tried with your original template but it didn't path parameters. Please notice that in my example, path parameters are put after path value, not in `method dictionary`.

In OTW, if you use `allParams` or `pathParams`, you won't get parameters which are defined like above example.

**Comment by dangthaison91** [comment]
Resolved by #29
